src/Filter.ts

```diff
@@ -0,0 +1,110 @@
+export interface FilterArgs {
+	[ name: string ]: Filter | RegExp | string | (() => boolean);
+}
+
+export default class Filter {
+	[index: string]: any;
+	type: string;
+	args: any[];
+
+	constructor();
+	constructor(filterArg: string);
+	constructor(filterArg: () => any);
+	constructor(filterArg: FilterArgs);
+	constructor(filterArg?: any) {
```

Comment:
could you use a union type instead of the multiple `constructor` declarations?

```ts
constructor(filterArg?: string | () => any | FilterArgs | any)
```

Additionally, not sure what our style guide says about this, but this definition will accept anything as a parameter. I can see the value of explicitly defining it this way for documentation purposes but we could just get away with

```ts
constructor(filterArgs?: any)
```